src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -434,32 +428,27 @@ impl<'b, 'tcx:'b> Resolver<'b, 'tcx> {
         }
 
         let name = xcdef.name;
-        let is_public = xcdef.vis == ty::Visibility::Public || parent.is_trait();
-
-        let mut modifiers = DefModifiers::empty();
-        if is_public {
-            modifiers = modifiers | DefModifiers::PUBLIC;
-        }
-        if parent.is_normal() {
-            modifiers = modifiers | DefModifiers::IMPORTABLE;
-        }
+        let vis = if parent.is_trait() { ty::Visibility::Public } else { xcdef.vis };
```

Comment:
`xcdef.vis` is still `PrivateExternal` for trait methods. I believe this is because visibilities of trait methods are not encoded in the metadata and the decoder considers items without encoded visibilities to be `PrivateExternal`.

Ideally, the decoder would give trait methods the same visibility as the trait itself, but that is orthogonal to this PR.